src/components/calcite-date-picker/utils.ts

```diff
@@ -36,10 +36,16 @@ export interface DateLocaleData {
 function getSupportedLocale(lang = "") {
   if (locales.indexOf(lang) > -1) {
     return lang;
-  } else {
-    const base = lang.split("-")[0];
-    return locales.indexOf(base) > -1 ? base : "en";
   }
+
+  lang = lang.toLowerCase();
+
+  if (lang.includes("-")) {
+    const langType = lang.split("-")[1];
+    const langId = lang.replace(langType, langType.toUpperCase());
```

Comment:
To simplify, this can be assigned to `lang` and you would no longer need the duplicate return statement in this block.